ckanext/switzerland/helpers/backend_helpers.py

```diff
@@ -190,3 +183,58 @@ def ogdch_resource_display_name(res):
                 logic.NotAuthorized, AttributeError):
             return ""
     return resource_display_name
+
+
+def dataset_display_name(package_or_package_dict):
+    """
+    monkey patched version of ckan.lib.helpers.dataset_display_name which
+    extracts the correct translation of the dataset name
+    """
+    name = dataset_display_name_orig(package_or_package_dict)
+    return get_localized_value_for_display(name)
+
+
+def organization_link(organization):
+    """
+    monkey patched version of ckan.lib.helpers.organization_link which extracts
+    the correct translation of the organization name
+    """
+    organization['title'] = get_localized_value_for_display(
+        organization['title'])
+    return organization_link_orig(organization)
+
+
+def group_link(group):
+    """
+    monkey patched version of ckan.lib.helpers.group_link which extracts the
+    correct translation of the group title
+    """
+    url = url_for(controller='group', action='read', id=group['name'])
+    title = group['title']
+    try:
+        # the group creation message contains str(dict), so we must parse the
+        # string to fix it. If the title is just a string, a ValueError is
+        # thrown.
+        title = ast.literal_eval(title)
+    except ValueError:
+        pass
+    title = get_localized_value_for_display(title)
+    return _link_to(title, url)
+
+
+def resource_link(resource_dict, package_id):
+    """
+    monkey patched version of ckan.lib.helpers.resource_link which extracts the
+    correct translation of the resource name
+    """
+    log.warning('resource_link')
```

Comment:
Do we need the logging here? If so, I would like it if you could add information so that this is more useful to us.
Considering this might be called a lot, I assume it is a relict of debugging? ;)

Reply:
Oops, this is indeed a relict of debugging. I will clean this up!